plugins/plugin-gdb/che-plugin-gdb-ide/src/main/java/org/eclipse/che/plugin/gdb/ide/configuration/GdbConfigurationPagePresenter.java

```diff
@@ -116,24 +119,38 @@ public void go(AcceptsOneWidget container) {
     }
 
     private void setHostsList() {
-        machineServiceClient.getMachines(appContext.getWorkspaceId()).then(new Operation<List<MachineDto>>() {
-            @Override
-            public void apply(List<MachineDto> machines) throws OperationException {
-                @SuppressWarnings("unchecked")
-                Promise<RecipeDescriptor>[] recipePromises = (Promise<RecipeDescriptor>[])new Promise[machines.size()];
-
-                for (int i = 0; i < machines.size(); i++) {
-                    String location = machines.get(i).getConfig().getSource().getLocation();
-                    String recipeId = getRecipeId(location);
-                    recipePromises[i] = recipeServiceClient.getRecipe(recipeId);
-                }
+        List<Machine> machines = getMachines();
+        if (machines.isEmpty()) {
+            return;
+        }
+
+        @SuppressWarnings("unchecked")
+        Promise<RecipeDescriptor>[] recipePromises = (Promise<RecipeDescriptor>[])new Promise[machines.size()];
+        for (int i = 0; i < machines.size(); i++) {
+            String location = machines.get(i).getConfig().getSource().getLocation();
+            String recipeId = getRecipeId(location);
+            recipePromises[i] = recipeServiceClient.getRecipe(recipeId);
+        }
+        setHostsList(recipePromises, machines);
+    }
+
+    private List<Machine> getMachines() {
+        List<Machine> machines = new ArrayList<>();
```

Comment:
@benoitf Sure, my thoughts are the following:

If I'm not mistaken the version of GWT that we use compiles both `ArrayList` and `LinkedList` into JavaScript array so as I mentioned previously there should not be a performance difference here. On the other hand I think it would be quite nice if we could avoid using of implementation specific code but use more generalized code instead. From that point of view for situations where we cannot know the initial size along with requirement of dynamical extension of collection it would be more appropriate to use LinkedList.

I would like to draw you attention that I said _production code_ to emphasize that the difference is insignificant for tests and other non-production environments while for environments where details are important it is worth of consideration. Phrase _recommendation for consideration_ means that I do realize that the benefit may be worthless and I only recommend to consider such option.

I'm curious to know what is your opinion about this?